server/src/middleware/usersMiddleware.ts

```diff
@@ -10,9 +10,13 @@ export const validateUser = async (
   const body = JSON.parse(event.body || "{}"); // Parse the JSON body
 
   try {
-    const existingUser = await UserService.getUserByEmail(body.email);
+    const existingEmail = await UserService.getUserByEmail(body.email);
 
-    if (existingUser) return { isValid: false, message: "כתובת מייל בשימוש!" };
+    if (existingEmail) return { isValid: false, message: "כתובת מייל בשימוש!" };
+
+    const existingPhone = await UserService.getUserByPhone(body.phone);
+
+    if (existingPhone) return { isValid: false, message: "מספר טלפון בשימוש!" };
```

Comment:
you should maybe do one function where you find  user where email matches given email or phone matches given phone number. This way you  only make one request to the database. and you also can loop through the results. and if anything matches then someone already has those values. You get me?

Reply:
The function should be generic btw, you should be able to pass in any conditions and find based off that. not just phoneor email

Reply:
making it genreic would make it require two requests

Reply:
also if i werent to make it genric and have it just accept both phone and email and return a user if any user matches either than the response will require some tweaking bc ill have to check what the user it returns has (email or phone) anyways so that i could send the proper message.

Reply:
anyways let me know which function you prefer